### ISSUE TITLE
Can't store values with '\n' character

### ISSUE THREAD

**Issue by diericx** [alicebob/miniredis#101]
There is an issue where you can't store any values that contain a \n character. This is causing some problems for me as I'm trying to marshal proto into redis (which separates values with a \n char). It works on a live redis server but not in miniredis. Here's a simple test I just wrote up:

```
func TestNewLine(t *testing.T) {
	redisServer := setup()
	defer teardown(redisServer)

	// Query to check that it exists in DB
	redisClient := redis.NewClient(&redis.Options{
		Addr:     redisServer.Addr(),
		Password: "", // no password set
		DB:       0,  // use default DB
	})

	err := redisClient.ZAdd("effects", redis.Z{
		Score:  float64(0),
		Member: "value",
	}).Err()
	if err != nil {
		t.Errorf("got unexpected error: %v", err)
	}

	err = redisClient.ZAdd("effects", redis.Z{
		Score:  float64(0),
		Member: "value2\n",
	}).Err()
	if err != nil {
		t.Errorf("got unexpected error: %v", err)
	}

	keys, _ := redisClient.ZScan("effects", 0, "*", 0).Val()
	fmt.Println(keys)
	if len(keys) != 4 {
		t.Errorf("got %v, want %v", len(keys), 4)
	}
}
```
Output:
```
[value 0]
--- FAIL: TestNewLine (0.00s)
    /Users/zachary/Documents/code/github.com/terrariumai/simulation/pkg/datacom/redis_test.go:776: got 2, want 4
FAIL
FAIL	github.com/terrariumai/simulation/pkg/datacom	0.033s
Error: Tests failed.
```
Looking at the print out, only the first ZAdd is working. Whats more interesting is that the second doesn't error out. The value is just not getting added.

**Comment by alicebob** [comment]
> There is an issue where you can't store any values that contain a \n character. This is causing some problems for me as I'm trying to marshal proto into redis (which separates values with a \n char). It works on a live redis server but not in miniredis. Here's a simple test I just wrote up:

Thanks for the clear test!, I'll have a look soon.

**Comment by alicebob** [comment]
Hi @diericx, #103 might fix the problem. The values were stored all-right, only the key match function didn't expect newlines. I'll merge the branch in a few days.

**Comment by alicebob** [comment]
I just merged it to master, @diericx.

Thanks!

**Comment by alicebob** [comment]
> This is still not working for me
> https://play.golang.org/p/JFW3Cqwvwhk

Interesting! I'll have a look soon, thanks!
(and sorry :)

**Comment by alicebob** [comment]
On Wed, Aug 28, 2019 at 10:18:47AM -0700, Zac Holland wrote:
> No worries, I know how it goes. I've got a workaround in place at the moment that just replaces /n characters with %n on the way in, and vice versa on the way out. 
> 
> Looking at your last PR, it might just be ZScan that isn't working (it wasn't changed). 

Hi,

sorry for the delay.
I hadn't released the last change as a proper version, I had only merged it to
master. I've now tagged and released it as v2.9.1, and the example works for me
now.
